### PR TITLE
Implement time zone methods in UCalendar

### LINKED ISSUE #73: Add time-zone related methods to `UCalendar`
There's been a request to expose functionality to get time zone offsets for a given time zone. The direct method, [`icu::TimeZone::getOffset()`](https://unicode-org.github.io/icu-docs/apidoc/released/icu4c/classicu_1_1TimeZone.html#afcbc1c48bf0b453b0123c4cb75d20e96), does not seem to have an equivalent in C, but a workaround can be cobbled together out of a few methods in `ucal.h`.

### PR DESCRIPTION
**Pull request by kpozin**
Fixes #73.

### CODE REVIEW

**Comment by filmil** [comment]
Looking.

File: rust_icu_ucal/src/lib.rs [hunk 1]
```diff
@@ -217,4 +331,86 @@ mod tests {
         let tz_version = super::get_tz_data_version().expect("get_tz_data_version works");
         assert!(re.is_match(&tz_version));
     }
+
+    #[test]
+    fn test_get_set_millis() -> Result<(), common::Error> {
+        let now = get_now();
+        let mut cal = UCalendar::new("America/New_York", "en-US", UCalendarType::UCAL_GREGORIAN)?;
+        assert!((now - cal.get_millis()?).abs() < 1000f64);
```
Comment (filmil) [inline rust_icu_ucal/src/lib.rs:339]: Care to explain this magic check?

File: rust_icu_ucal/src/lib.rs [hunk 2]
```diff
@@ -217,4 +331,86 @@ mod tests {
...
+
+        let date = now + 17.0;
```
Comment (filmil) [inline rust_icu_ucal/src/lib.rs:341]: Declare a clearly named constant instead of magic 17?

File: rust_icu_ucal/src/lib.rs [hunk 3]
```diff
@@ -217,4 +331,86 @@ mod tests {
...
+        cal.set_millis(date)?;
+        assert_eq!(cal.get_millis()?, date);
+        Ok(())
+    }
+
+    #[test]
+    fn test_set_date() -> Result<(), common::Error> {
+        // Thu May 07 2020 21:00:00 GMT-0400 (US Eastern Daylight Time)
+        let time_a: UDate = 1588899600000.0;
```
Comment (filmil) [inline rust_icu_ucal/src/lib.rs:350]: Why not use parse time?  I think calendar supports it?

File: rust_icu_ucal/src/lib.rs [hunk 4]
```diff
@@ -217,4 +331,86 @@ mod tests {
...
+        assert_eq!(cal.get(UCalendarDateFields::UCAL_MILLISECOND)?, 898);
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_offsets_and_daylight_time() -> Result<(), common::Error> {
+        let mut cal = UCalendar::new("America/New_York", "en-US", UCalendarType::UCAL_GREGORIAN)?;
+
+        let expected_zone_offset: i32 = -5 * 60 * 60 * 1000;
```
Comment (filmil) [inline rust_icu_ucal/src/lib.rs:398]: 5 hours vs 1 hour?  Units? Document?  Here and below.

File: rust_icu_ucal/src/lib.rs [hunk 5]
```diff
@@ -217,4 +331,86 @@ mod tests {
...
+        let expected_dst_offset: i32 = 1 * 60 * 60 * 1000;
+
+        cal.set_date_time(2020, 4 /* May */, 7, 21, 0, 0)?;
```
Comment (filmil) [inline rust_icu_ucal/src/lib.rs:401]: Nit: would it be worthwhile to define separate enum types for each of these params?  They are kind of easy to mix up at the moment. 0 for January is especially weird.

File: rust_icu_ucal/src/lib.rs [hunk 6]
```diff
@@ -217,4 +331,86 @@ mod tests {
...
+    #[test]
+    fn test_set_date_time() -> Result<(), common::Error> {
+        // Thu May 07 2020 21:26:55.898 GMT-0400 (US Eastern Daylight Time)
+        let time_a: UDate = 1588901215898.0;
+        // Mon May 04 2020 21:00:00.898 GMT-0400 (US Eastern Daylight Time)
+        let time_b: UDate = 1588640400898.0;
+
+        let mut cal = UCalendar::new("America/New_York", "en-US", UCalendarType::UCAL_GREGORIAN)?;
+        cal.set_millis(time_a)?;
+        cal.set_date_time(2020, 4 /* May */, 4, 21, 0, 0)?;
```
Comment (filmil) [inline rust_icu_ucal/src/lib.rs:371]: Consider also adding a test that is always about a date which is 1 year ahead of whatever is "today".  This way, if things somehow change such that we'd be broken in the future, we'd know 1 year in advance.

Comment (filmil) [reply rust_icu_ucal/src/lib.rs]: > Do you mean something like a test that confirms that America/New_York still follows DST one year in the future?

This, in fact.  But, at your option.

Comment (kpozin) [reply rust_icu_ucal/src/lib.rs]: We don't have any code (to my knowledge) that relies on time zone rules remaining static, so let's avoid creating tests that might break unhelpfully :)

**Pull request merged**